File: playground/02-SelectionSort/selection_sort.py
def find_smallest(array):
  """Return the index for the smallest number in array"""
  smallest = array[0]
  smallest_index = 0

  for i in range(1, len(array)):
    if array[i] < smallest:
      smallest = array[i]
      smallest_index = i
  return smallest_index

test_array = [20, 23, 33, 4, 5, 1, 3, 1, 3, 0]
assert find_smallest(test_array) == 9

File: playground/02-SelectionSort/selection-sort.ts
const testArray = [20, 23, 33, 4, 5, 1, 3, 1, 3, 0, -1];

const findSmallest = (array: number[]): number => {
	let smallestNumber = array[0];
	let smallestIndex = 0;

	for (let index = 0; index < array.length; index++) {
		if (array[index] < smallestNumber) {
			smallestNumber = array[index];
			smallestIndex = index;
		}
	}

	return smallestIndex;
};

if (findSmallest(testArray) === 10) {
	console.log('Correct');
} else {
	throw new Error('Incorrect');
}
